### PR TITLE
Regenerate profile SAP_ALL

### LINKED ISSUE #4782: Regenerate SAP_ALL profile after pulling some authorization object
Hi guys,

I have added a new authorization object (AO) into our repository. This AO is successfully tracked by abapgit which is great and it can be pulled into a new system. However profile SAP_ALL does not contain this new AO.

Issue description: Profile SAP_ALL is automatically regenerated when transport with a new AO arrives into a target system and field SAP_ALL_GENERATION is set to ON in the table PRGN_CUST. However this is not the case after doing abapgit pull. 

Proposal: Add support for regeneration of profile SAP_ALL using abapgit. Something similar is already doing the report AGR_REGENERATE_SAP_ALL, which regenerates SAP_APP for every client in the system, so its logic can be taken as an example.

What do you think about it?

Thanks,
Juraj

### PR DESCRIPTION
**Pull request by juriadasek**
Regenerate profile SAP_ALL after pulling some authorization object using abapgit. Please review, my first PR here :) 

Closes #4782

### CODE REVIEW

File: src/objects/zcl_abapgit_object_suso.clas.abap [hunk 1]
```diff
@@ -154,6 +156,24 @@ CLASS ZCL_ABAPGIT_OBJECT_SUSO IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD regenerate_sap_all.
+
+    DATA: lt_e071  TYPE STANDARD TABLE OF e071,
+          lt_e071k TYPE STANDARD TABLE OF e071k.
+
+    lt_e071 = VALUE #( ( pgmid = 'R3TR' object = 'SUSO' ) ).
+
```
Comment (mbtools) [inline src/objects/zcl_abapgit_object_suso.clas.abap:165]:
```abap
ls_e071-pgmid = 'R3TR'.
ls_e071-object = ms_item-obj_type.
ls_e071-obj_name = ms_item-obj_name.
INSERT ls_e071 INTO TABLE et_071.
```

**Review by mbtools — CHANGES_REQUESTED** [review]
Small change to be compatible with 702 but good otherwise :-)

**Comment by mbtools** [comment]
Can you pls check if the delete method removes the AO from SAP_ALL. If not, add the regenerate there, too.

**Comment by juriadasek** [comment]
@mbtools Do you know when this delete method is called? I couldn't find any 'delete' on UI. I just want to test it and make sure that everything is ok.

**Comment by mbtools** [comment]
'Advanced > Uninstall' will call delete for each object of the repo

**Comment by jfilak** [comment]
This method https://github.com/abapGit/abapGit/blob/85878b972a775918ddf674fef41143dc94d09ffa/src/objects/zcl_abapgit_object_suso.clas.abap#L182

File: src/objects/zcl_abapgit_object_suso.clas.abap (hunk 1 above)
Comment (juriadasek) [reply src/objects/zcl_abapgit_object_suso.clas.abap:169]: Fixed. I used that older syntax.

**Comment by jfilak** [comment]
BTW: don't we need to update the changelong?

**Comment by flaiker** [comment]
> BTW: don't we need to update the changelong?

Usually the changelog is updated for each release, not for every pull request.

**Comment by larshp** [comment]
welcome to abapGit as a contributor 🎉

**Pull request merged**